lib/events/auditlog.go

```diff
@@ -952,55 +952,47 @@ func (l *AuditLog) SearchSessionEvents(ctx context.Context, req SearchSessionEve
 
 // StreamSessionEvents implements [SessionStreamer].
 func (l *AuditLog) StreamSessionEvents(ctx context.Context, sessionID session.ID, startIndex int64) (chan apievents.AuditEvent, chan error) {
-	l.log.Debugf("StreamSessionEvents(%v)", sessionID)
+	l.log.WithField("session_id", string(sessionID)).Debug("StreamSessionEvents()")
 	e := make(chan error, 1)
 	c := make(chan apievents.AuditEvent)
 
-	tarballPath := filepath.Join(l.playbackDir, string(sessionID)+".stream.tar")
-	downloadCtx, cancel := l.createOrGetDownload(tarballPath)
-
-	// Wait until another in progress download finishes and use its tarball.
-	if cancel == nil {
-		l.log.Debugf("Another download is in progress for %v, waiting until it gets completed.", sessionID)
-		select {
-		case <-downloadCtx.Done():
-		case <-l.ctx.Done():
-			e <- trace.BadParameter("audit log is closing, aborting the download")
-			return c, e
-		}
-	} else {
-		defer cancel()
-	}
-
-	rawSession, err := os.OpenFile(tarballPath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0o640)
+	rawSession, err := os.CreateTemp(l.playbackDir, string(sessionID)+".stream.tar.*")
```

Comment:
The following flow where tmp file is created and imedialty removed is quite hacky from reader perspective:
```
rawSession, err := os.CreateTemp(l.playbackDir, string(sessionID)+".stream.tar.*")
err := os.Remove(rawSession.Name())
// Using removed file fd
l.UploadHandler.Download(l.ctx, sessionID, rawSession)
```

Can we at least add a comment describing this behavior ?